src/ctr_exit.c

```diff
@@ -159,6 +160,14 @@ void do_exit_command()
 	close_fd(&console_socket_fd);
 	close_fd(&attach_pipe_fd);
 
+	/*
+	 * We don't want the exit command to be reaped by the parent conmon
+	 * as that would prevent double-fork from doing its job.
+	 * Unfortunately, that also means that any new subchildren from
+	 * still running processes could also get lost
+	 */
+	prctl(PR_SET_CHILD_SUBREAPER, 0);
+
```

Comment:
Underneath this there's a block
```c
	if (exit_pid) {
		int ret, exit_status = 0;

		/*
		 * Make sure to cleanup any zombie process that the container runtime
		 * could have left around.
		 */
		do {
			int tmp;

			exit_status = 0;
			ret = waitpid(-1, &tmp, 0);
			if (ret == exit_pid)
				exit_status = get_exit_status(tmp);
		} while ((ret < 0 && errno == EINTR) || ret > 0);

		if (exit_status)
			_exit(exit_status);

		return;
	}

```

I believe this block was to cleanup these grandchildren. what processes could we leak in podman cleanup if we drop the subreaper @mheon @giuseppe

Reply:
Assuming container restart, there are a few fork/execs - c/storage (reexec code), CNI plugins, and then a fresh Conmon (possible I'm forgetting a few but these are the ones I can recall). All of these could potentially leak if Podman were killed at the right time, though the only one where I'd say this is likely is CNI, which takes a long while to execute and gives a decently long window for this race.